tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs

```diff
@@ -91,20 +90,24 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
-        public bool UpdateRules(IEnumerable<RuleData[]> res)
+        public bool UpdateRules(IEnumerable<RuleData> res)
         {
-            var finalRuleDatas = MergeRuleDatas(res);
+            if (res == null || res.Count() == 0)
+            {
+                return false;
+            }
 
+            var finalRuleDatas = MergeRuleDatas(res);
             var encoded = encoder.Encode(finalRuleDatas, new List<Obj>(), false);
-
             var ret = wafNative.UpdateRuleData(ruleHandle.Value, encoded.RawPtr);
+            Log.Debug("{number} rules have been updated and status is {status}", finalRuleDatas.Count, ret);
             return ret == DDWAF_RET_CODE.DDWAF_OK;
         }
 
-        internal static List<object> MergeRuleDatas(IEnumerable<RuleData[]> res)
+        internal static List<object> MergeRuleDatas(IEnumerable<RuleData> res)
```

Comment:
Does "datas" exist? 🙈

Reply:
I'm not sure I understand 😅 , `Datas `is an array contained by each `RuleData` sent by backend, slightly confusing indeed^^ It's such as:
```
{
    "rules_data": [
        {
            "data": [
                {
                    "expiration": 32522625382,
                    "value": "1.1.1.1"
                },
                {
                    "value": "52.80.198.1"
                },
                {
                    "expiration": 1663146835,
                    "value": "44.201.225.88"
                }
            ],
            "id": "blocked_ips",
            "type": "ip_with_expiration"
        }
    ]
}
```

Reply:
wow ok thx fixed in f7f395e14e5bd4ff126f045189695a42c7fdbe72